TileStache/Goodies/VecTiles/transform.py

```diff
@@ -3093,3 +2972,155 @@ def normalize_medical_kind(shape, properties, fid, zoom):
+    def __init__(self, values):
+        self.values = values
+
+    def match(self, other):
+        return other in self.values
+
+    def __repr__(self):
+        return repr(self.value)
+
+
+_KEY_TYPE_LOOKUP = {
+    'int': int,
+    'float': float,
+}
+
+def _parse_kt(key_type):
+    kt = key_type.split("::")
+
+    type_key = kt[1] if len(kt) > 1 else None
+    fn = _KEY_TYPE_LOOKUP.get(type_key, str)
+
+    return (kt[0], fn)
+
+
+class CSVMatcher(object):
+    def __init__(self, fh):
+        keys = None
+        types = []
+        rows = []
+
+        self.static_any = _AnyMatcher()
+        self.static_none = _NoneMatcher()
+        self.static_some = _SomeMatcher()
+
+        # CSV - allow whitespace after the comma
+        reader = csv.reader(fh, skipinitialspace=True)
+        for row in reader:
+            if keys is None:
+                target_key = row.pop(-1)
+                keys = []
+                for key_type in row:
+                    key, typ = _parse_kt(key_type)
+                    keys.append(key)
+                    types.append(typ)
+
+            else:
+                target_val = row.pop(-1)
+                for i in range(0, len(row)):
+                    row[i] = self._match_val(row[i], types[i])
+                rows.append((row, target_val))
+
+        self.keys = keys
+        self.rows = rows
+        self.target_key = target_key
+
+    def _match_val(self, v, typ):
+        if v == '*':
+            return self.static_any
+        if v == '-':
+            return self.static_none
+        if v == '+':
+            return self.static_some
+        if isinstance(v, str) and ';' in v:
+            return _SetMatcher(set(v.split(';')))
+        return _ExactMatcher(typ(v))
+
+    def __call__(self, properties):
+        vals = [properties.get(k) for k in self.keys]
+        for row, target_val in self.rows:
+            if all([a.match(b) for (a, b) in zip(row, vals)]):
+                return (self.target_key, target_val)
+
+        return None
+
+
+def csv_match_properties(ctx):
+    """
+    Add or update a property on all features which match properties which are
+    given as headings in a CSV file.
+    """
+
+    feature_layers = ctx.feature_layers
+    zoom = ctx.tile_coord.zoom
+    source_layer = ctx.params.get('source_layer')
+    start_zoom = ctx.params.get('start_zoom', 0)
+    end_zoom = ctx.params.get('end_zoom')
+    target_value_type = ctx.params.get('target_value_type')
+    matcher = ctx.resources.get('matcher')
+
+    assert source_layer, 'csv_match_properties: missing source layer'
+    assert matcher, 'csv_match_properties: missing matcher resource'
+
+    if zoom < start_zoom:
+        return None
+
+    if end_zoom is not None and zoom > end_zoom:
+        return None
+
+    layer = _find_layer(feature_layers, source_layer)
+    if layer is None:
+        return None
+
+    def _type_cast(v):
+        if target_value_type == 'int':
+            return int(v)
+        return v
```

Comment:
Yeah, it'll do `str` and `int` now, and I couldn't think of any others we needed right now. We don't really have many `float` properties.